### PR TITLE
Compile manifests instead of interpreting them

### PR DESCRIPTION
**Pull request by hartbit**
Notice that this change requires modifications to `bootstrap` and `ManifestLoader` to make sure the compiled manifest can find the dynamic `PackageDescription` runtime library.

### CODE REVIEW

**Comment by hartbit** [comment]
@swift-ci smoke test

**Comment by hartbit** [comment]
@swift-ci smoke test

**Comment by hartbit** [comment]
@swift-ci smoke test

**Comment by hartbit** [comment]
@aciidb0mb3r Do you have an idea what is causing the failures on Linux? I haven't been able to reproduce them in Docker.

**Comment by shahmishal** [comment]
@swift-ci smoke test Linux

File: Utilities/bootstrap [hunk 1]
```diff
@@ -441,6 +441,14 @@ def build_swiftpm_with_cmake(args):
 
     build_with_cmake(args, cmake_flags, args.project_root, args.bootstrap_dir)
 
+    if platform.system() == 'Darwin':
```
Comment (aciidb0mb3r) [inline Utilities/bootstrap]: It's better to set the install name while building the PackageDescription libraries. You can do so by adding this in the Sources/PackageDescription/CMakeLists.txt file:

```
    if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
      target_link_options(PD${PACKAGE_DESCRIPTION_VERSION} PRIVATE
        "SHELL:-Xlinker -install_name -Xlinker @rpath/PackageDescription.dylib")
    endif()
```

**Comment by aciidb0mb3r** [comment]
For linux, can you try reproducing by installing the toolchain in some other directory instead of system's /usr? I think it's somehow looking for the swift runtime object at the default path instead of the toolchain 🤔

File: Utilities/bootstrap (hunk 1 above)
Comment (hartbit) [reply Utilities/bootstrap]: Great! I initially trying playing with CMake's INSTALL_NAME_DIR, but it seems it only works when installing.

**Comment by hartbit** [comment]
But that wouldn't be causing the exact same issue for Linux self hosted?

**Comment by hartbit** [comment]
@swift-ci smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test self hosted

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test self hosted

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by fortmarek** [comment]
Hi, I could not find the reason behind this change - is it a question of performance or is there something else?

**Comment by hartbit** [comment]
It’s a question of portability. The interpreter is usually much less stable on other platforms than the compiler.

**Pull request merged**